src/segment_winreg.go

```diff
@@ -32,11 +32,17 @@ func (wr *winreg) enabled() bool {
 
 	var err error
 	wr.Value, err = wr.env.getWindowsRegistryKeyValue(registryPath, registryKey)
-	if len(fallback) > 0 && (err != nil || len(wr.Value) == 0) {
-		wr.Value = fallback
-		return true
+
+	if err != nil {
```

Comment:
Can you do an early return here? `if err == nil`, then we can continue the logic in this block unindented below.

Reply:
Gotcha, like this:

```
	if err == nil {
		return true
	}

	if len(fallback) > 0 {
		wr.Value = fallback
		return true
	}

	return false

```

Feels like there's almost a razor here - i.e. if you find yourself writing "else" to go through codeblocks that contain return, look again you can likely optimize the early-out structure.

Reply:
Exactly. I also didn't invent this strategy, it comes from [object calisthenics](https://gist.github.com/bobuss/6534934). While not every rule is always applicable or relevant, it still produces very readable code when you keep it in mind.

Reply:
Interesting, thanks.  I was always a "single return path" guy when I was engineering full-time, it's fun to challenge that!